### ISSUE TITLE
Performance issue with md-icon and md-svg-icon

### ISSUE THREAD

**Issue by TBxy** [angular/material#8687]
**Actual behavior**:
It is very slow when i use `md-svg-icon`.

I load the `mdi.svg` from https://materialdesignicons.com with a`$mdIconProvider`, but the same issue exists with other (smaller) iconsets.
Pre-fetching the icons did also not help.

**CodePen**
http://codepen.io/tburgherr/pen/BzodNm

Large delay when adding SVG icons, oke for font-icons and avatar images.

**Angular Versions**: 
-  Angular Version: 1.5.3
-  Angular Material Version: 1.0.9

**Additional Information**:
-  Browser Type: Chrome
-  Browser Version: 47.0.2526.106 (64-bit)
-  OS: Linux

------------------------------------------------------------

Is there something I do wrong, or should I use font icons if many icons are needed?

**Comment by DavidGalvanDom** [comment]
+1

**Comment by crisbeto** [comment]
It's taking a while, because RawGit is taking 5+ sec to respond, in addition to the file being 700kb. For example, [try visiting the URL directly](http://cdn.rawgit.com/nkoterba/material-design-iconsets/master/iconsets/mdi-icons.svg) and you'll see exactly how much time it takes to download the icons. Also bear in mind that RawGit are a free service and they don't give any guarantees about speed and reliability. You might have better luck hosting the file yourself.

**Comment by clshortfuse** [comment]
I don't see a speed issue myself. Even with hard-cache and reload

    137 requests | 1.4 MB transferred | Finished 6.09s | DOMContentLoaded 1.44s | Load 1.46s

Even if it set Chrome to throttle to 2G, the render time is .43 seconds (Load - DOMContentLoaded)

I also see you're running Chrome v47. The latest is v50/v51. Maybe it's an older issue with Chrome?

**Comment by ThomasBurleson** [comment]
This is also reported on v1.0.9 (a patch release). Please confirm that a performance issues exists with v1.1.0-rc.5 or higher; and we can reopen at that time.

**Comment by DavidGalvanDom** [comment]
I have this issue with IE 11 low performance  using  ng-repeat     <md-icon class="md-toggle-icon"  md-svg-icon="dots-vertical" ></md-icon>

**Comment by TBxy** [comment]
I updated my codepen (http://codepen.io/tburgherr/pen/BzodNm) to *v1.1.0-rc.5*.
I also tested it with a newer chrome (51.0.2704.84, 64-bit) and with firefox (43.0), both on Linux.

**The issue is still the same**.

Even if it takes 5 seconds the first time to load the SVGs it should be much faster the second time (after toggling the SVG checkbox). For my code I use a local copy of the SVGs, but it is slow there as well.

And it does not perform better if **ng-repeat is not used** (see: http://codepen.io/tburgherr/pen/mEEppe [demo without ng-repeat]).
